lib/models/user_model.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class UserModel {
+  String id; 
+  UserType type;
+  String password;
+  String phone;
+  bool isOnline;
+  Map<String,String> vehicle = {
+  "make": "",
+  "model": "",
+  "licensePlate": ""
+  };
+  GeoPoint location;
+
+  UserModel({
+    required this.id,
+    required this.type,
+    required this.phone,
+    required this.isOnline,
+    required this.vehicle,
+    required this.location,
+    required this.password
+  });
+
+  factory UserModel.fromFireStore(DocumentSnapshot doc) {
```

Comment:
although it looks good idea, but this means your model is tightly coupled to firebase, so in case you wanna connect to another db you have to update model as well